modules/apps/dynamic-data-mapping/dynamic-data-mapping-service/src/main/java/com/liferay/dynamic/data/mapping/internal/upgrade/v2_0_1/UpgradeDDMFormFieldValidation.java

```diff
@@ -0,0 +1,190 @@
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS
+ * FOR A PARTICULAR PURPOSE. See the GNU Lesser General Public License for more
+ * details.
+ */
+
+package com.liferay.dynamic.data.mapping.internal.upgrade.v2_0_1;
+
+import com.liferay.portal.kernel.dao.jdbc.AutoBatchPreparedStatementUtil;
+import com.liferay.portal.kernel.exception.PortalException;
+import com.liferay.portal.kernel.json.JSONArray;
+import com.liferay.portal.kernel.json.JSONFactory;
+import com.liferay.portal.kernel.json.JSONObject;
+import com.liferay.portal.kernel.upgrade.UpgradeProcess;
+import com.liferay.portal.kernel.util.PortalUtil;
+import com.liferay.portal.kernel.util.StringUtil;
+import com.liferay.portal.kernel.util.Validator;
+
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+
+/**
+ * @author Lino Alves
+ */
+public class UpgradeDDMFormFieldValidation extends UpgradeProcess {
+
+	public UpgradeDDMFormFieldValidation(JSONFactory jsonFactory) {
+		_jsonFactory = jsonFactory;
+	}
+
+	@Override
+	protected void doUpgrade() throws Exception {
+		try (PreparedStatement ps1 = connection.prepareStatement(
+				"select structureId, definition from DDMStructure where " +
+					"classNameId = ? ");
+			PreparedStatement ps2 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructure set definition = ? where " +
+						"structureId = ?");
+			PreparedStatement ps3 = connection.prepareStatement(
+				"select structureVersionId, definition from " +
+					"DDMStructureVersion where structureId = ?");
+			PreparedStatement ps4 =
+				AutoBatchPreparedStatementUtil.concurrentAutoBatch(
+					connection,
+					"update DDMStructureVersion set definition = ? where " +
+						"structureVersionId = ?")) {
+
+			ps1.setLong(
+				1,
+				PortalUtil.getClassNameId(
+					"com.liferay.dynamic.data.mapping.model.DDMFormInstance"));
+
+			try (ResultSet rs = ps1.executeQuery()) {
+				while (rs.next()) {
+					String definition = rs.getString("definition");
+
+					ps2.setString(1, updateValidation(definition));
+
+					long structureId = rs.getLong("structureId");
+
+					ps2.setLong(2, structureId);
+
+					ps2.addBatch();
+
+					ps3.setLong(1, structureId);
+
+					try (ResultSet rs2 = ps3.executeQuery()) {
+						while (rs2.next()) {
+							definition = rs2.getString("definition");
+
+							ps4.setString(1, updateValidation(definition));
+
+							long structureVersionId = rs2.getLong(
+								"structureVersionId");
+
+							ps4.setLong(2, structureVersionId);
+
+							ps4.addBatch();
+						}
+					}
+				}
+			}
+
+			ps2.executeBatch();
+
+			ps4.executeBatch();
+		}
+	}
+
+	protected String updateValidation(String definitionJSON)
+		throws PortalException {
+
+		JSONObject definition = _jsonFactory.createJSONObject(definitionJSON);
+
+		JSONArray fields = definition.getJSONArray("fields");
+
+		for (int i = 0; i < fields.length(); i++) {
+			JSONObject field = fields.getJSONObject(i);
```

Comment:
@linolaoj, please add the suffix `JSONObject` to all variable names which are JSON Objects.